zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/UserTaskJobBasedHandler.java

```diff
@@ -129,8 +141,15 @@ public void flush(final TaskEntity entity, final BatchRequest batchRequest) {
     final var updateFields = getUpdatedFields(entity);
     final var taskEntityId = entity.getId();
     final var processInstanceKey = entity.getProcessInstanceId();
+
+    final boolean previousVersionRecord = refersToPreviousVersionRecord(entity.getKey());
+
     batchRequest.upsertWithRouting(
-        indexName, taskEntityId, entity, updateFields, processInstanceKey);
+        indexName,
+        previousVersionRecord ? String.valueOf(entity.getKey()) : taskEntityId,
+        entity,
+        updateFields,
+        previousVersionRecord ? String.valueOf(entity.getKey()) : processInstanceKey);
```

Comment:
🔧 Here the same. Maybe we can extract this logic even to some helper, with clear API/docs